src/libstd/sys/unix/thread.rs

```diff
@@ -125,8 +126,10 @@ impl Thread {
     }
 
     pub fn sleep(dur: Duration) {
+        let secs = dur.as_secs();
+        assert!(secs <= (u64::MAX >> 1), "secs number is too big");
```

Comment:
I think it still would be more clear if you did something like:

```
let secs = dur.as_secs() as libc::time_t;
assert!(secs.is_positive(), "...");
```

Your current approach won’t catch the cases where `time_t` is `i32` (some versions of windows, mips linux etc)